packages/dds/tree/src/core/tree/visitDelta.ts

```diff
@@ -128,23 +123,25 @@ function visitFieldMarks(
 
 function visitModify(
 	index: number,
-	modify: ModifyLike,
+	modify: Delta.HasModifications,
 	visitor: DeltaVisitor,
 	config: PassConfig,
 ): boolean {
 	let containsMovesOrDeletes = false;
-	visitor.enterNode(index);
 	// Note that the `in` operator return true for properties that are present on the object even if they
 	// are set to `undefined. This is leveraged here to represent the fact that the value should be set to
 	// `undefined` as opposed to leaving the value untouched.
-	if (config.applyValueChanges && "setValue" in modify) {
+	if (config.applyValueChanges && Object.prototype.hasOwnProperty.call(modify, "setValue")) {
+		visitor.enterNode(index);
```

Comment:
The reason I didn't want the enter/exit outside the if statements is that `visitModify` is now being called on Delta marks whether or not they have changes in them (which would lead to having pairs or enter/exits with nothing in between).

It is indeed a slight perf hit if both are set, but that should be rare in practice (because nodes typicaly represent either values or objects).
Still, it was a bit weird so I updated the code to avoid either issue and look less confusing.